### PR TITLE
Ported service tests to ServiceTestCase (fixes #158)

### LINKED ISSUE #158: Port service tests to ServiceTestCase
Follow up on #41 and PR #155.

1) Convert

   test/service/test_pagination.py
   test/service/test_request_builder.py
   test/service/test_response.py

   to ServiceTestCase and fix any breakage in them. If breakage is severe, @skip the test and file a ticket.

2) Remove `Dockerfile`, `docker.compose*.yml`, shell scripts and other support files that are only needed for the old way of running the tests inside a pair of Docker containers.

3) Move test data (json files, tsvs etc into separate `test/service/data` directory.

4) Convert the test/service/Makefile to run all unit tests with `python -m unittest discover` or similar.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/AZUL-118)
┆Project Name: azul
┆Issue Number: AZUL-118

### PR DESCRIPTION
**Pull request by GPelayo**
**Issue**: https://github.com/DataBiosphere/azul/issues/158

### CODE REVIEW

File: test/service/test_request_builder.py [hunk 1]
```diff
@@ -5,16 +5,19 @@
...
 
 logging.config.fileConfig(os.path.join(os.path.dirname(azul.service.config.__file__), 'logging.conf'))
 logger = logging.getLogger("dashboardService")
 
+DATA_DIRECTORY = os.path.join(SERVICE_TESTS_FOLDER, 'data')
+REQUEST_CONFIG_FILEPATH = os.path.join(DATA_DIRECTORY, 'test_request_config.json')
 
-class MyTestCase(unittest.TestCase):
-
+@unittest.skip
```
Comment (hannes-ucsc) [inline test/service/test_request_builder.py]: @unittest.skip("http://github.org/ref/to/the/issue_that_will_fix_this")

File: test/service/test_request_builder.py [hunk 2]
```diff
@@ -5,16 +5,19 @@
...
+from service import WebServiceTestCase, SERVICE_TESTS_FOLDER
 from azul.service.responseobjects.elastic_request_builder import ElasticTransformDump as EsTd
 import azul.service.config
 from azul import config
 
 logging.config.fileConfig(os.path.join(os.path.dirname(azul.service.config.__file__), 'logging.conf'))
 logger = logging.getLogger("dashboardService")
 
+DATA_DIRECTORY = os.path.join(SERVICE_TESTS_FOLDER, 'data')
+REQUEST_CONFIG_FILEPATH = os.path.join(DATA_DIRECTORY, 'test_request_config.json')
```
Comment (hannes-ucsc) [inline test/service/test_request_builder.py]: Class attribute.

File: test/service/test_response.py [hunk 3]
```diff
@@ -3,13 +3,15 @@
 import json
 import difflib
 import unittest
+from service import WebServiceTestCase
 from azul.service.responseobjects.hca_response_v5 import KeywordSearchResponse, FileSearchResponse
 import os
 
-base_path = os.path.dirname(os.path.abspath(__file__))
+DATA_FOLDER_FILEPATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
```
Comment (hannes-ucsc) [inline test/service/test_response.py]: Class attribute.

File: test/service/test_response.py [hunk 4]
```diff
@@ -3,13 +3,15 @@
...
 
 
-class MyTestCase(unittest.TestCase):
+@unittest.skip
+class MyTestCase(WebServiceTestCase):
```
Comment (hannes-ucsc) [inline test/service/test_response.py]: TestResponse

File: test/service/test_response.py [hunk 5]
```diff
@@ -3,13 +3,15 @@
...
+from service import WebServiceTestCase
 from azul.service.responseobjects.hca_response_v5 import KeywordSearchResponse, FileSearchResponse
 import os
 
-base_path = os.path.dirname(os.path.abspath(__file__))
+DATA_FOLDER_FILEPATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
 
-class MyTestCase(unittest.TestCase):
+@unittest.skip
```
Comment (hannes-ucsc) [inline test/service/test_response.py]: @unittest.skip("http://github.org/ref/to/the/issue_that_will_fix_this")

File: Makefile [hunk 6]
```diff
@@ -28,7 +28,8 @@ clean:
 	for d in lambdas terraform; do $(MAKE) -C $$d clean; done
 
 test:
-	$(MAKE) -C test/service testme
+	$(MAKE) -C test test
```
[the remark this answers is not in the document]
Comment (GPelayo) [reply Makefile]: Done

File: test/Makefile [hunk 7]
```diff
@@ -0,0 +1,2 @@
+test:
```
[the remark this answers is not in the document]
Comment (GPelayo) [reply test/Makefile]: The only phony targets should be "all" and "test" right?

File: test/service/test_request_builder.py [hunk 8]
```diff
@@ -13,7 +14,12 @@
 logger = logging.getLogger("dashboardService")
 
 
-class MyTestCase(unittest.TestCase):
+class RequestBuilderTestCase(WebServiceTestCase):
```
[the remark this answers is not in the document]
Comment (GPelayo) [reply test/service/test_request_builder.py]: Done

File: test/service/__init__.py [hunk 9]
```diff
@@ -10,15 +10,16 @@ class setUpModule():
 
 
 class WebServiceTestCase(AzulTestCase):
+    service_tests_folder = os.path.dirname(os.path.realpath(__file__))
```
[the remark this answers is not in the document]
Comment (GPelayo) [reply test/service/__init__.py]: Done

File: test/indexer/__init__.py [hunk 10]
```diff
@@ -9,7 +9,7 @@
 
 
 class IndexerTestCase(AzulTestCase):
-
+    indexer_tests_folder = os.path.dirname(os.path.realpath(__file__))
```
[the remark this answers is not in the document]
Comment (GPelayo) [reply test/indexer/__init__.py]: Done

File: test/service/__init__.py [hunk 11]
```diff
@@ -10,15 +10,16 @@ class setUpModule():
...
     data_loader = None
 
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
         cls.data_loader = ElasticsearchFakeDataLoader(
```
[the remark this answers is not in the document]
Comment (GPelayo) [reply test/service/__init__.py]: Done

**Comment by hannes-ucsc** [comment]
This patch fixes running the test from `test`. The culprit was more hard-coded logging configuration `test_request_builder.py`. Don't forget to remove logging.conf.

```diff
Index: test/service/test_request_builder.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/service/test_request_builder.py	(revision 146630a8313084e6e847d5f8a87a241279b57d23)
+++ test/service/test_request_builder.py	(date 1533763636000)
@@ -7,11 +7,9 @@
 import unittest
 from service import WebServiceTestCase
 from azul.service.responseobjects.elastic_request_builder import ElasticTransformDump as EsTd
-import azul.service.config
 from azul import config
 
-logging.config.fileConfig(os.path.join(os.path.dirname(azul.service.config.__file__), 'logging.conf'))
-logger = logging.getLogger("dashboardService")
+logger = logging.getLogger(__name__)
 
 
 class RequestBuilderTestCase(WebServiceTestCase):
Index: test/service/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/service/__init__.py	(revision 146630a8313084e6e847d5f8a87a241279b57d23)
+++ test/service/__init__.py	(date 1533760416000)
@@ -1,14 +1,9 @@
 import os
-import logging
 
 from shared import AzulTestCase
 from service.data_generator.fake_data_utils import ElasticsearchFakeDataLoader
 
 
-class setUpModule():
-    logging.basicConfig(level=logging.INFO)
-
-
 class WebServiceTestCase(AzulTestCase):
     service_tests_folder = os.path.dirname(os.path.realpath(__file__))
     data_loader = None
Index: test/indexer/test_hca_indexer.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/indexer/test_hca_indexer.py	(revision 146630a8313084e6e847d5f8a87a241279b57d23)
+++ test/indexer/test_hca_indexer.py	(date 1533763745000)
@@ -187,9 +187,6 @@
         """
         We submit two different bundles for the same specimen. What happens?
         """
-        logs = logging.getLogger('azul.indexer')
-        logs.disabled = False
-
         spec1_pack = self._get_data_files(self.spec1_bundle[0])
         spec2_pack = self._get_data_files(self.spec2_bundle[0])
         specimen_list = [(self.spec1_bundle, spec1_pack),
@@ -219,7 +216,6 @@
                     if "There was a conflict with document" in log_msg:
                         num_hits += 1
                 self.assertEqual(3, num_hits)
-        logs.disabled = True
 
         def check_specimen_merge(es_results):
             specimen_uuids = []
Index: test/service/data_generator/fake_data_utils.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/service/data_generator/fake_data_utils.py	(revision 146630a8313084e6e847d5f8a87a241279b57d23)
+++ test/service/data_generator/fake_data_utils.py	(date 1533760162000)
@@ -6,11 +6,6 @@
 import os
 
 logger = logging.getLogger(__name__)
-logger.setLevel(logging.DEBUG)
-
-
-class setUpModule():
-    logging.basicConfig(level=logging.WARNING)
 
 
 class FakerSchemaGenerator(object):
diff --git src/azul/service/config/logging.conf src/azul/service/config/logging.conf
deleted file mode 100644
index b44fec308363057d05b44687909628b94a80405f..0000000000000000000000000000000000000000
GIT binary patch
literal 0
Hc$@<O00001

```

File: test/Makefile (hunk 7 above)
[the remark this answers is not in the document]
Comment (GPelayo) [reply test/Makefile]: Done

File: test/indexer/test_hca_indexer.py [hunk 12]
```diff
@@ -37,9 +38,8 @@ def setUpClass(cls):
         cls.spec2_bundle = ("56a338fe-7554-4b5d-96a2-7df127a7640b",
                             "2018-03-29T153507.198365Z")
 
-    @staticmethod
-    def _get_data_files(filename, updated=False):
-        data_prefix = "data/"
+    def _get_data_files(self, filename, updated=False):
+        data_prefix = f"{os.path.dirname(os.path.realpath(__file__))}/data/"
```
Comment (hannes-ucsc) [inline test/indexer/test_hca_indexer.py]: os.path.join() would probably look more natural here.

File: test/indexer/test_indexer.py [hunk 13]
```diff
@@ -1,6 +1,6 @@
 import unittest
-from app import get_bundles, get_file, flatten, look_file, es_check, \
-    write_es, write_index, es
+# from app import get_bundles, get_file, flatten, look_file, es_check, \
```
Comment (hannes-ucsc) [inline test/indexer/test_indexer.py]: Remove completely.

File: test/indexer/test_indexer.py [hunk 14]
```diff
@@ -10,7 +10,7 @@
 
 # bb_host = "https://"+os.environ['BLUE_BOX_ENDPOINT']
 # in_host = "https://"+os.environ['INDEXER_ENDPOINT']
-es_host = os.environ['ES_ENDPOINT']
+# es_host = os.environ['ES_ENDPOINT']
```
Comment (hannes-ucsc) [inline test/indexer/test_indexer.py]: Ditto

File: test/indexer/test_indexer.py [hunk 15]
```diff
@@ -154,6 +154,7 @@
                    "analysis,json|computational_method": "None"}]
 
 
+@unittest.skip('')
```
Comment (hannes-ucsc) [inline test/indexer/test_indexer.py]: Needs ticket.

File: test/indexer/test_projects.py [hunk 16]
```diff
@@ -48,6 +47,7 @@ def setUpClass(cls):
     def tearDownClass(cls):
         super().tearDownClass()
 
+    @unittest.skip
```
Comment (hannes-ucsc) [inline test/indexer/test_projects.py]: Needs ticket.

File: test/service/test_request_builder.py [hunk 17]
```diff
@@ -5,15 +5,19 @@
...
-logger = logging.getLogger("dashboardService")
+logger = logging.getLogger(__name__)
 
 
-class MyTestCase(unittest.TestCase):
+class TestRequestBuilder(WebServiceTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.data_directory = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
```
Comment (hannes-ucsc) [inline test/service/test_request_builder.py]: Yeah. This looks better than f"" sttring method.

File: test/indexer/test_projects.py (hunk 16 above)
Comment (GPelayo) [reply test/indexer/test_projects.py]: This line shouldn't have been committed. My mistake.

File: test/indexer/test_hca_indexer.py (hunk 12 above)
Comment (GPelayo) [reply test/indexer/test_hca_indexer.py]: Done

File: test/indexer/test_indexer.py (hunk 13 above)
Comment (GPelayo) [reply test/indexer/test_indexer.py]: Done. Also removed the unused os import

File: test/indexer/test_indexer.py (hunk 14 above)
Comment (GPelayo) [reply test/indexer/test_indexer.py]: Done

File: test/indexer/test_indexer.py (hunk 15 above)
Comment (GPelayo) [reply test/indexer/test_indexer.py]: Ticket made at https://github.com/DataBiosphere/azul/issues/177.

File: test/service/test_request_builder.py (hunk 17 above)
Comment (GPelayo) [reply test/service/test_request_builder.py]: OK. I will follow this format.

**Comment by hannes-ucsc** [comment]
Thank you. I tweaked the commit title a bit.

**Pull request merged**